azure/scope/identity.go

```diff
@@ -219,6 +226,33 @@ func (p *AzureCredentialsProvider) GetAuthorizer(ctx context.Context, resourceMa
 	return autorest.NewBearerAuthorizer(spt), nil
 }
 
+// GetClientID returns the Client ID associated with the AzureCredentialsProvider's Identity.
+func (p *AzureCredentialsProvider) GetClientID() string {
+	return p.Identity.Spec.ClientID
+}
+
+// GetClientSecret returns the Client Secret associated with the AzureCredentialsProvider's Identity.
+// NOTE: this only works if the Identity references a Service Principal Client Secret.
+// If using another type of credentials, such a Certificate, we return an empty string.
```

Comment:
Clarification: How does cloud provider get the credentials when using other type of credentials (eg. certificate)? Is it expected to use a managed identity for such cases?

Reply:
The answer to your question is coming here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1521